lib/core/src/lib/viewer/components/viewer.component.ts

```diff
@@ -180,10 +210,14 @@ export class ViewerComponent<T> implements OnDestroy, OnInit, OnChanges {
 
     private closeViewer = true;
     private keyDown$ = fromEvent<KeyboardEvent>(document, 'keydown');
+    private isDialogVisible: boolean = false;
+    public downloadPromptTimer: any;
```

Comment:
Please set types for both timer to avoid using `any`

Reply:
Updated types to number. However, because of this, also had to change from using just settimeout() (which returns NodeJS.Timeout) to window.setTimeout() (which returns number, the timeout ID). Refer below changes -
 

- [downloadPrompt](https://github.com/Alfresco/alfresco-ng2-components/pull/8428/files#diff-8edc4da1fb1bec4de886be767b73f594e2c448e23cf2858260ba7f60acce88b3R364)
- [downloadPomptReminder](https://github.com/Alfresco/alfresco-ng2-components/pull/8428/files#diff-8edc4da1fb1bec4de886be767b73f594e2c448e23cf2858260ba7f60acce88b3R394)

Reply:
I think you can try this approach https://stackoverflow.com/a/56239226 it makes your `setTimeout` safe and platform independent, also you could still use NodeJS.Timeout then